core/src/main/java/com/datastax/oss/driver/internal/core/cql/QueryTraceFetcher.java

```diff
@@ -50,6 +52,17 @@
       DriverConfigProfile configProfile) {
     this.tracingId = tracingId;
     this.session = session;
+
+    ConsistencyLevel regularConsistency =
+        configProfile.getConsistencyLevel(CoreDriverOption.REQUEST_CONSISTENCY);
+    ConsistencyLevel traceConsistency =
+        configProfile.getConsistencyLevel(CoreDriverOption.REQUEST_TRACE_CONSISTENCY);
+    this.configProfile =
+        (traceConsistency == regularConsistency)
+            ? configProfile
+            : configProfile.withConsistencyLevel(
+                CoreDriverOption.REQUEST_CONSISTENCY, traceConsistency);
```

Comment:
This creates a new profile for every trace retrieval. It should be fine since tracing has a performance hit anyway.